astropy/units/format/ogip.py

```diff
@@ -86,14 +97,10 @@ def _generate_unit_names():
         deprecated_units = ["Crab", "mCrab"]
         for unit in deprecated_units:
             deprecated_names.add(unit)
-        functions = [
-            "log", "ln", "exp", "sqrt", "sin", "cos", "tan", "asin",
-            "acos", "atan", "sinh", "cosh", "tanh",
-        ]  # fmt: skip
-        for name in functions:
+        for name in cls._functions:
             names[name] = name
```

Comment:
We can one-line simple dictionary updates

```suggestion
        names |= dict(zip(cls._functions, cls._functions))
```

Reply:
The `_generate_unit_names()` methods of all the `Generic` subclasses can be simplified quite a lot. Should I expand this pull request?

Reply:
I think that would be in scope!

Reply:
I added various small changes to the `_generate_unit_names()`. If you think they are beyond the scope here then just drop the third commit and merge the first two.

Reply:
LGTM.